atomic_reactor/__init__.py

```diff
@@ -32,6 +35,28 @@ def format(self, record):
         return super(ArchFormatter, self).format(record)
 
 
+class EncodedStream(object):
+    def __init__(self, fileno, encoding):
+        self.binarystream = fdopen(dup(fileno), 'wb')
```

Comment:
We dup() the file descriptor here because the `file` returned by `os.fdopen()` will call `close()` on its file descriptor when it is deleted -- and so will `sys.stderr`.